pkg/driver/helmdriver.go

```diff
@@ -69,13 +72,37 @@ func (d *helmDriver) Install(ctx context.Context,
 	if err != nil {
 		return fmt.Errorf("loading helm chart %s: %w", name, err)
 	}
+	// If no target namespace provided read chart values to find namespace
+	if namespace == "" {
+		val, exists := helmChart.Values["defaultNamespace"]
+		if exists {
+			namespace = val.(string)
+		} else {
+			// Fall back case of assuming its default
+			namespace = "default"
+		}
+	}
+
+	// Update values with imagePullSecrets
+	// If no secret values we should still continue as it could be case of public registry or local registry
+	secretvals, err := d.secretAuth.GetSecretValues(ctx, namespace)
+	if err == nil {
+		for key, val := range secretvals {
+			values[key] = val
+		}
+	}
```

Comment:
This is highly unusual for go code.

So much so, that I'd rather see this:

```suggestion
	secretvals, err := d.secretAuth.GetSecretValues(ctx, namespace)
	if err != nil {	
		// possibly a log.Warnf("something smoething")
		secretvals = nil
		// Continue, even though we failed to load secrets (or some words that makes sense to you)
	}

	// (remove me, this is for edification) The nil value will just skip right through...
	for key, val := range secretvals {
		values[key] = val
	}

```
Most people are just trained to see a call, then an error check with `err != nil`. If you mix it up and call `err == nil` on them, you should have a very good reason, and probably a comment that attempts to alert them. In this case, we do have an error, just a non-fatal one, and we can log that or otherwise deal with it, while keeping to the more common pattern of `err != nil`.

Reply:
this was done in latest commit